src/discord/mod.rs

```diff
@@ -221,13 +223,24 @@ pub struct ErrorResponse {
     internal_error: String,
 }
 
+// TODO: some kind of impl From<&str> for ErrorResponse that returns an error response with "Internal error, sorry."
+//       as the user facing error and the &str as the internal one?
+//       or maybe From<E: Error> that behaves similarly
+
 impl ErrorResponse {
     fn new<S1: Into<String>, S2: Display>(user_facing_error: S1, internal_error: S2) -> Self {
         Self {
             user_facing_error: user_facing_error.into(),
             internal_error: internal_error.to_string(),
         }
     }
+
+    fn new_internal(internal_error: impl Display) -> Self {
+        Self {
+            user_facing_error: "Internal error, sorry.".to_string(),
+            internal_error: internal_error.to_string(),
+        }
+    }
```

Comment:
it would be good to do a sweep of replacing some annoying code with this

but actually i think the `ErrorResponse` thing has basically been a failure overall and it would be nice to come up with a better pattern entirely :\